### PR TITLE
Avoid the view for mkldnn case in 1D convolution

### LINKED ISSUE #68034: 1D convolution is broken for mkldnn tensors
## 🐛 Bug

1D convolution is broken for mkldnn tensors and a badly-worded error is thrown for this case.

## To Reproduce
```python
import torch

input = torch.randn(2, 3, 10).to_mkldnn()
weight = torch.randn(3, 3, 3).to_mkldnn()
bias = torch.randn(3).to_mkldnn()
output = torch.nn.functional.conv1d(input, weight, bias)
```
```
RuntimeError: opaque tensors do not have strides
```

## Expected behavior

Either the correct output is returned for mkldnn tensor inputs or a proper error is thrown indicating the lack of support for 1D convolution with mkldnn tensors.

## Additional Context
The problem occurs when trying to view 1D spatial input / weight as 2D. This is done for other backends (e.g. cuDNN) that don't support 1D spatial input directly. However, the [mkldnn convolution docs](https://oneapi-src.github.io/oneDNN/v1.0/dev_guide_convolution.html) indicate that 1D spatial inputs are supported directly, so it should be an easy fix to avoid the view for the mkldnn case.
https://github.com/pytorch/pytorch/blob/a1d733ae8ca4bbad8fa22a4a532b67915114ae93/aten/src/ATen/native/Convolution.cpp#L889-L897

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @albanD @mruberry @walterddr @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @VitalyFedyunin

### PR DESCRIPTION
**Pull request by yanbing-j**
Fixes https://github.com/pytorch/pytorch/issues/68034

### CODE REVIEW

File: test/test_mkldnn.py [hunk 1]
```diff
@@ -309,6 +309,12 @@ def test_conv2d_legacy_jit_model(self):
                 conv2d(x),
                 conv2d_loaded(x.to_mkldnn()).to_dense())
 
+    def test_conv1d_functional(self):
```
Comment (jbschlosser) [inline test/test_mkldnn.py:312]: Please add a comment explaining why this test is here, maybe linking back to the issue.

I also suggest verifying the output shape as an additional sanity check.

**Review by jbschlosser — COMMENTED** [review]
Thanks for the fix! :) Added a few nitpicky comments but it's looking good!

Note: to pass the lint check, the now unused import `expectedFailure` in test_nn.py should be removed.

File: aten/src/ATen/native/Convolution.cpp [hunk 2]
```diff
@@ -827,11 +827,9 @@ ConvBackend select_conv_backend(
   check_shape_forward(input, weight.sizes(), bias, params);
 
   // Expand 1d -> 2d.
```
[the remark this answers is not in the document]
Comment (yanbing-j) [reply aten/src/ATen/native/Convolution.cpp:829]: Done.

File: test/test_mkldnn.py (hunk 1 above)
Comment (yanbing-j) [reply test/test_mkldnn.py:314]: Done.

**Review by jbschlosser — APPROVED** [review]
LGTM thanks for the fix!

**Pull request closed without merging**